From 1cb42951e84776c7f32e6d2f243b9f41c167584a Mon Sep 17 00:00:00 2001
From: Fabse <fabse@uwmc.info>
Date: Mon, 28 Mar 2016 00:46:46 +0200
Subject: [PATCH] Minecart drop name workaround

This is a workaround for MC-68446 (or similar).
In Survival Minecraft this bug only shows in minecart drops, so this is the only thing fixed here.

diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
index a649ae7..f0d503c 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
@@ -168,7 +168,7 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
         if (this.world.getGameRules().getBoolean("doEntityDrops")) {
             ItemStack itemstack = new ItemStack(Items.MINECART, 1);
 
-            if (this.getName() != null) {
+            if (this.hasCustomName()) { // Paper - Only set item name if custom name is set. Workaround for MC-68446
                 itemstack.c(this.getName());
             }
 
-- 
2.7.4

